src/asciidoc/docs/guidelines/frontendUploading.adoc

```diff
@@ -0,0 +1,108 @@
+
+This method enables the file to be uploaded directly from the Koji’s frontend to the Koji CDN, for optimal performance.
+Other methods, such as `uploadFile`, would require the file to be uploaded to the project’s backend server and then to the CDN.
+
+To upload files with `generateSignedUploadRequest`, you configure the backend to acquire a presigned URL, which grants temporary access for uploading a file.
+Then, you configure the frontend to retrieve a presigned URL from the backend and use it to upload a file.
+
+== Prerequisites
+
+This guide covers advanced topics.
+It is suggested that you are familiar with the following topics before proceeding.
+
+* React and ES6 basics.
+* The Koji editor, <<instant-remixing#, remix process>>, and the <<withkoji-vcc-package,@withkoji/vcc>> package.
+* https://expressjs.com/[Express.js] web framework.
+* Backend communication and the <<withkoji-database-package,@withkoji/database-sdk>> package.
+* Performing HTTP requests (this example uses https://github.com/axios/axios[axios]).
+
+== Configuring the backend
+
+To upload from the frontend, you first need to acquire a presigned URL by using the Koji database SDK.
+
+In a backend service for your Koji project, set up an Express.js server with a POST method to the `/get-upload-url` endpoint.
+From this endpoint, call the following function.
+
+[source, javascript]
+----
+import Database from '@withkoji/database';
+
+const uploadImage = async (req, res) => {
+  try {
+    const { filename } = req.body;
+
+    const db = new Database({
+      projectId: res.locals.KOJI_PROJECT_ID,
+      projectToken: res.locals.KOJI_PROJECT_TOKEN,
+    });
+
+    const data = await db.generateSignedUploadRequest(filename);
+
+    res.status(200).json({ data });
+  } catch (err) {
+    console.log('err', err);
+    res.status(500).json({ err: err.toString() });
+  }
+};
+
+export default uploadImage;
+----
+
+== Configuring the frontend
+
+From your frontend service, call the `/get-upload-url` endpoint to retrieve the presigned URL.
+
+[source,javascript]
+----
+const { data: { data: { signedRequest, url } } = {} } = await axios({
+  method: 'post',
+  url: `${serviceMap.backend}/get-upload-url`,
+  data: {
+    filename: file.name,
+  },
+  headers: {
+    'Content-Type': 'application/json',
+  },
+});
+----
+
+Create form data and include the presigned URL data and the file object to be uploaded.
+
+[source,javascript]
+----
+// Build form data to send to s3
+const formData = new FormData();
+
+// Attach each of the request fields to the form
+Object.keys(signedRequest.fields).forEach((key) => {
+  formData.append(key, signedRequest.fields[key]);
+});
+
+// Append file last
+formData.append('file', file);
+----
+
+Post this data to the URL, and then check for a successful status (since there is no response data).
```

Comment:
It could be helpful to show even a basic use case of what this would look like. Something like 

```
const image = new window.Image()
image.onload = () => {
  document.body.append(image);
}
image.src = signedRequest.url;
```